src/main/java/org/apache/commons/text/similarity/IntersectionResult.java

```diff
@@ -0,0 +1,166 @@
+ */
+public class IntersectionResult {
+    /**
+     * The size of set A.
+     */
+    private final int sizeA;
+    /**
+     * The size of set B.
+     */
+    private final int sizeB;
+    /**
+     * The size of the intersection between set A and B.
+     */
+    private final int intersection;
+
+    /**
+     * Create the results for an intersection between two sets.
+     *
+     * @param sizeA the size of set A ({@code |A|})
+     * @param sizeB the size of set B ({@code |B|})
+     * @param intersection the size of the intersection of A and B (<code>|A &#8745; B|</code>)
+     * @throws IllegalArgumentException if the sizes are negative or the intersection is greater
+     * than the minimum of the two set sizes
+     */
+    public IntersectionResult(final int sizeA, final int sizeB, final int intersection) {
+        if (sizeA < 0) {
+            throw new IllegalArgumentException("Set size |A| is not positive: " + sizeA);
+        }
+        if (sizeB < 0) {
+            throw new IllegalArgumentException("Set size |B| is not positive: " + sizeB);
+        }
+        if (intersection < 0 || intersection > Math.min(sizeA, sizeB)) {
+            throw new IllegalArgumentException("Invalid intersection of |A| and |B|: " + intersection);
+        }
+        this.sizeA = sizeA;
+        this.sizeB = sizeB;
+        this.intersection = intersection;
+    }
+
+    /**
+     * Get the size of set A.
+     *
+     * @return |A|
+     */
+    public int getSizeA() {
+        return sizeA;
+    }
+
+    /**
+     * Get the size of set B.
+     *
+     * @return |B|
+     */
+    public int getSizeB() {
+        return sizeB;
+    }
+
+    /**
+     * Get the size of the intersection between set A and B.
+     *
+     * @return <code>|A &#8745; B|</code>
+     */
+    public int getIntersection() {
+        return intersection;
+    }
+    /**
+     * Get the size of the union between set A and B.
+     *
+     * @return <code>|A &#8746; B|</code>
+     */
+    public long getUnion() {
+        return (long) sizeA + sizeB - intersection;
+    }
+
+    /**
+     * Gets the Jaccard index. The Jaccard is the intersection divided by the union.
+     *
+     * <pre><code>|A &#8745; B| / |A &#8746; B| </code></pre>
+     *
+     * <p>This implementation defines the result as zero if there is no intersection,
+     * even when the union is zero to avoid a {@link Double#NaN} result.</p>
+     *
+     * @return the Jaccard index
+     * @see <a href="https://en.wikipedia.org/wiki/Jaccard_index">Jaccard index</a>
+     */
+    public double getJaccardIndex() {
+        return intersection == 0 ? 0.0 : (double) intersection / getUnion();
+    }
+
+    /**
+     * Gets the S&#248;rensen-Dice coefficient. The coefficient is twice the size of the intersection
+     * divided by the size of both sets.
+     *
+     * <pre>
+     * <code>2|A &#8745; B| / (|A| + |B|) </code>
+     * </pre>
+     *
+     * <p>This is also known as the F1 score.
```

Comment:
Thanks. I also noticed that but had stopped firing up insignificant changes to GitHub. It is not detected by checkstyle and I ran `mvn javadoc` OK. It would have to be a JDK < 1.8 to break building the javadoc IIUC.